File: backend/src/main/java/com/enterprise/callrecorder/model/CallRecording.java
package com.enterprise.callrecorder.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * JPA Entity for Call Recordings
 */
@Entity
@Table(
    name = "call_recordings",
    indexes = {
        @Index(name = "idx_device_id", columnList = "device_id"),
        @Index(name = "idx_phone_number", columnList = "phone_number"),
        @Index(name = "idx_status", columnList = "status"),
        @Index(name = "idx_created_at", columnList = "created_at")
    }
)
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CallRecording {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 255)
    private String deviceId;

    @Column(length = 255)
    private String callLogId;

    @Column(nullable = false, length = 20)
    private String phoneNumber;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    private CallType callType;

    @Column(nullable = false)
    private Long startedAt;

    private Long endedAt;

    private Long duration;

    // File information
    @Column(length = 255, unique = true)
    private String fileName;

    private Long fileSize;

    @Column(length = 255)
    private String storageKey;

    // Status
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    @Builder.Default
    private CallStatus status = CallStatus.RECORDING;

    @Column(columnDefinition = "integer default 0")
    @Builder.Default
    private Integer uploadAttempts = 0;

    @Column(columnDefinition = "TEXT")
    private String lastUploadError;

    // Audit
    @Column(nullable = false, updatable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    private LocalDateTime uploadedAt;

    private LocalDateTime deletedAt;
}

File: android/src/main/java/com/enterprise/callrecorder/model/CallRecording.kt
package com.enterprise.callrecorder.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.*

/**
 * Entité CallRecording pour la base de données
 */
@Entity(tableName = "call_recordings")
data class CallRecording(
    @PrimaryKey(autoGenerate = true)
    val id: Long? = null,
    
    // 🔴 Identification
    val deviceId: String,
    val callLogId: String? = null,
    val phoneNumber: String,
    
    // 🔴 Type et timing
    val callType: CallType,
    val startedAt: Long,
    val endedAt: Long? = null,
    val duration: Long? = null,
    
    // 🔴 Fichier
    val fileName: String? = null,
    val fileSize: Long? = null,
    val storageKey: String? = null,
    
    // 🔴 Statut
    val status: CallStatus = CallStatus.RECORDING,
    val uploadAttempts: Int = 0,
    val lastUploadError: String? = null,
    
    // 🔴 Timestamps
    val createdAt: Long = System.currentTimeMillis(),
    val uploadedAt: Long? = null,
    val deletedAt: Long? = null
)

/**
 * Type d'appel
 */
enum class CallType {
    INCOMING,
    OUTGOING,
    MISSED,
    REJECTED
}

/**
 * Statuts de l'enregistrement
 */
enum class CallStatus {
    RECORDING,          // En cours d'enregistrement
    PROCESSING,         // Traitement post-appel
    PENDING_UPLOAD,     // En attente d'upload
    UPLOADING,          // Upload en cours
    UPLOADED,           // Uploadé avec succès
    UPLOAD_FAILED,      // Erreur d'upload
    DELETED             // Supprimé
}
